tomcat/src/main/java/org/apache/catalina/startup/SessionManager.java

```diff
@@ -0,0 +1,54 @@
+package org.apache.catalina.startup;
+
+import ch.qos.logback.core.spi.LifeCycle;
+import common.http.Session;
+import org.apache.catalina.Manager;
+
+public class SessionManager implements Manager, LifeCycle {
+
+    private static final Sessions sessions = new Sessions();
+    private boolean isStarted = false;
+
+    public SessionManager() {
+        start();
+    }
+
+    @Override
+    public void add(Session session) {
+        sessions.add(session.getId(), session);
+    }
+
+    @Override
+    public Session findSession(String id) {
+        return sessions.has(id);
+    }
+
+    @Override
+    public void remove(Session session) {
+        sessions.remove(session.getId());
+    }
+
+    @Override
+    public void start() {
+        if (!isStarted) {
+            isStarted = true;
+        }
+    }
+
+    @Override
+    public void stop() {
+        if (isStarted) {
+            sessions.clear();
+            isStarted = false;
+        }
+    }
+
+    @Override
+    public boolean isStarted() {
+        return isStarted;
+    }
```

Comment:
`LifeCycle`은 어떤 의도로 사용하셨나요?

Reply:
주어진 manager 클래스에 주석으로 아래와 같이 lifecycle을 구현해야한다고 적혀있네요!

/**
 * A <b>Manager</b> manages the pool of Sessions that are associated with a
 * particular Container.  Different Manager implementations may support
 * value-added features such as the persistent storage of session data,
 * as well as migrating sessions for distributable web applications.
 * <p>
 * In order for a <code>Manager</code> implementation to successfully operate
 * with a <code>Context</code> implementation that implements reloading, it
 * must obey the following constraints:
 * <ul>
 * <li>Must implement <code>Lifecycle</code> so that the Context can indicate
 *     that a restart is required.
 * <li>Must allow a call to <code>stop()</code> to be followed by a call to
 *     <code>start()</code> on the same <code>Manager</code> instance.
 * </ul>
 *
 * @author Craig R. McClanahan
 */